pkg/crc/cache/cache.go

```diff
@@ -25,12 +25,13 @@ type Cache struct {
 }
 
 type VersionMismatchError struct {
+	BinaryName      string
 	ExpectedVersion string
 	CurrentVersion  string
 }
 
 func (e *VersionMismatchError) Error() string {
-	return fmt.Sprintf("expected: %s but got: %s", e.ExpectedVersion, e.CurrentVersion)
+	return fmt.Sprintf("%s version mismatch: %s expected but %s found in the cache. Run 'crc setup' to update the cache.", e.BinaryName, e.ExpectedVersion, e.CurrentVersion)
```

Comment:
Mentioning `crc setup` here (in the cache package) feels a bit out of place, and it's also not really relevant when using json output I would say. Might be better in `crc start` as 
```
var e VersionMismatchError;
if errors.As(err, &e) {
    logging.Errorf("%s. Run 'crc setup' to update the cache.", e.Error())
}
```

Or am I overthinking this?

Reply:
I could to that but this check is called in the preflight package, so it won't be that closer to the start command. 

I can remove the "Run crc setup" part. This only happens when I update crc, I guess I can write a better check on top of all commands.

Reply:
This version mismatch is only part of preflight checks and right now all the preflight checks should be fixed by running the `crc setup` so I think adding it in `crc start` or even every commands is bit over doing.

Reply:
I removed the howto fix it part.